tester.abi_contract always calls last function

```
abi_contract_code = """
def mul2(a):
    return(a * 2)

def returnten():
    return(10)
"""

def test_abi_contract():
    s = tester.state()
    c = s.abi_contract(abi_contract_code)
    assert c.mul2(3) == [6]
    assert c.returnten() == [10]
```

```
=========================================================================== FAILURES ============================================================================
_______________________________________________________________________ test_abi_contract _______________________________________________________________________

    def test_abi_contract():
        s = tester.state()
        c = s.abi_contract(abi_contract_code)
>       assert c.mul2(3) == [6]
E       assert [10] == [6]
E         At index 0 diff: 10 != 6
E         Use -v to get the full diff

tests/test_contracts.py:1118: AssertionError
```

--- a/pyethereum/tester.py
+++ b/pyethereum/tester.py
@@ -76,11 +76,11 @@ class state():
                     "Contract code empty"
                 sig = serpent.mk_signature(code)
                 sig = sig[sig.find('[')+1:sig.rfind(']')].split(',')
-                for i, s in enumerate(sig):
+                for funid, s in enumerate(sig):
                     fun = s[:s.find(':')].strip()
                     funsig = s[s.find(':')+1:].strip()
 
-                    def kall_factory(fun, funsig):
+                    def kall_factory(funid, fun, funsig):
 
                         def kall(*abi, **kwargs):
                             if len(funsig) != len(abi):
@@ -94,10 +94,10 @@ class state():
                             return _state.send(kwargs.get('sender', k0),
                                                self.address,
                                                kwargs.get('value', 0),
-                                               funid=i, abi=abi)
+                                               funid=funid, abi=abi)
                         return kall
 
-                    vars(self)[fun] = kall_factory(fun, funsig)
+                    vars(self)[fun] = kall_factory(funid, fun, funsig)
 
         return _abi_contract(me, code, sender, endowment)
 

--- a/tests/test_contracts.py
+++ b/tests/test_contracts.py
@@ -1103,6 +1103,22 @@ def test_prevhashes():
     assert o2 == t2
 
 
+abi_contract_code = """
+def mul2(a):
+    return(a * 2)
+
+def returnten():
+    return(10)
+"""
+
+
+def test_abi_contract():
+    s = tester.state()
+    c = s.abi_contract(abi_contract_code)
+    assert c.mul2(3) == [6]
+    assert c.returnten() == [10]
+
+
 # test_evm = None
 # test_sixten = None
 # test_returnten = None
@@ -1135,3 +1151,4 @@ def test_prevhashes():
 # test_types_in_functions = None
 # test_more_infinites = None
 # test_prevhashes = None
+# test_abi_contract = None

